packages/wrangler/src/dispatch-namespaces/fetcher.ts

```diff
@@ -0,0 +1,113 @@
+import { fetch, Response } from "miniflare";
+import { UserError } from "../errors";
+import type { WorkerDefinition } from "../dev-registry";
+import type { Request } from "miniflare";
+
+export const EXTERNAL_DISPATCH_WORKER_NAME =
+	"__WRANGLER_EXTERNAL_DISPATCH_WORKER";
+
+const HEADER_SCRIPT_NAME = "X-Miniflare-Dispatch-Script-Name";
+export const HEADER_URL = "X-Miniflare-Dispatch-URL";
+export const HEADER_OUTBOUND_PROXY_URL =
+	"X-Miniflare-Dispatch-Outbound-Proxy-URL";
+export const HEADER_PARAMETERS = "X-Miniflare-Dispatch-Outbound-Parameters";
+export const HEADER_CF_BLOB = "X-Miniflare-Dispatch-Cf-Blob";
+
+// This should probably be doing all the clever proxy class stuff in miniflare.ts
+export const generateExternalDispatchWorkerScript = (
+	parameters: string[] = []
+) => `
+const HEADER_SCRIPT_NAME = ${JSON.stringify(HEADER_SCRIPT_NAME)};
+const HEADER_URL = ${JSON.stringify(HEADER_URL)};
+const HEADER_OUTBOUND_PROXY_URL = ${JSON.stringify(HEADER_OUTBOUND_PROXY_URL)};
+const HEADER_PARAMETERS = ${JSON.stringify(HEADER_PARAMETERS)};
+const HEADER_CF_BLOB = ${JSON.stringify(HEADER_CF_BLOB)};
+
+const ALLOWED_PARAMETERS = ${JSON.stringify(parameters)};
+
+export default function (env) {
+    return {
+			get(name, args, options) {
+				return {
+					fetch(request) {
+						request = new Request(request);
+						request.headers.set(HEADER_SCRIPT_NAME, name);
+						request.headers.set(HEADER_URL, request.url);
+						request.headers.set(HEADER_PARAMETERS, JSON.stringify(Object.fromEntries(Object.entries(options?.outbound ?? {}).filter(([key]) => ALLOWED_PARAMETERS.includes(key)))));
+						request.headers.set(HEADER_CF_BLOB, JSON.stringify(request.cf));
+						return env.FETCHER.fetch(request);
+					}
+				}
+			}
+		}
+}
+`;
+
+export function generateDispatchFetcher(
+	namespace: string,
+	outbound?: { service: string },
+	workerDefinitions: Record<string, WorkerDefinition> = {}
+) {
+	return async (request: Request): Promise<Response> => {
+		const scriptName = request.headers.get(HEADER_SCRIPT_NAME);
+		const dispatcheeTarget = Object.entries(workerDefinitions).find(
+			([name, workerDefinition]) =>
+				name === scriptName && workerDefinition.dispatchNamespace === namespace
+		)?.[1];
+
+		if (!dispatcheeTarget) {
+			return new Response(
+				`[wrangler] Couldn't find \`wrangler dev\` session for "${scriptName}" in namespace "${namespace}" to proxy to`,
```

Comment:
i don't remember specifically what these errors surface as in edgeworker, i think it's either `internal error` or `script not found`. Should we try and match that behavior or is being more verbose here better?